lib/puppet/provider/zone/solaris.rb

```diff
@@ -37,43 +53,42 @@ def self.instances
   # file or string.
   def configure
 
-    if @resource[:archive].nil? && @resource[:zonecfg_export].nil?
+    if @resource[:zonecfg_archive].nil? && @resource[:zonecfg_export].nil?
       raise Puppet::Error, "No configuration resource is defined."
     end
 
-    command = String.new
+    command = "#{command(:cfg)} -z #{@resource[:name]}"
     if @resource[:clone]
       if !@resource[:zonecfg_export]
         raise Puppet::Error, "A zone configuration must be defined to
         clone a zone."
       end
-      command = "#{command(:cfg)} -z #{@resource[:name]} -f #{@resource[:zonecfg_export]}"
+      command << " -f #{@resource[:zonecfg_export]}"
     else
       unless @resource[:zonecfg_export].nil? || @resource[:zonecfg_export].empty?
         begin
           file = File.open(@resource[:zonecfg_export], "rb")
-          str = file.read.gsub(/[\n]\n*\s*/, "; ")
+          exported_cfg = file.read.gsub(/[\n]\n*\s*/, "; ")
         rescue
```

Comment:
If File.open fails to open '@resource[:zonecfg_export]' treat the argument as a string containing the contents / commands to zonecfg. This is confusing to users.

Pull in "22126555 Agent reporting syntax error while creating a zone"